app/controllers/concerns/user_invitation.rb

```diff
@@ -62,6 +72,22 @@ def invite_new_user(email:, login: nil, first_name: nil, last_name: nil)
   end
 
   ##
+  # Sends a new invitation to the user with a new token.
+  #
+  # @param user_id [Integer] ID of the user to be re-invited.
+  # @return [Token] The new token used for the invitation.
+  def reinvite_user(user_id)
+    token = Token.find_by user_id: user_id, action: token_action
+
+    token.destroy! if token
+    new_token = Token.create! user_id: user_id, action: token_action
```

Comment:
I'm using the bang methods because it would really be **exception**al situations where they (destroy and create) fail. Accordingly the exceptions are not handled either resulting in a 500 which is what I would return anyway in those cases as you can't recover from those.